### DISCUSSION TITLE
When full screen-ing App1, is there a way to auto un-full screen App2?

### DISCUSSION THREAD

**Discussion by anonimitoraf** [asmvik/yabai#1258] [Q&A] [answered]
Similar to the i3 behavior where if you full screen app 1 (and app 2 is currently full screen'd), app 2 gets un-full screen'd automatically

**Comment by asmvik** [comment, marked as answer]
Are you talking about *fullscreen zoom* or *native macOS fullscreen* ?

Either way you can make a script that queries the windows and checks for this, then toggles it off, before making the new window fullscreen.

**Reply by Demianeen** [reply]
@koekeishiya, came with the same issue. How do you query for the event of entering full screen with yabai?

**Reply by asmvik** [reply]
The following will query all native-fullscreen windows and un-fullscreen them:


`yabai -m query --windows | jq 'map(select(."is-native-fullscreen") | .id) | join(" ")' | tr -d '"' | xargs -n1 -I{} yabai -m window {} --toggle native-fullscreen`

**Reply by asmvik** [reply]
You can target a specific application like so:

`yabai -m query --windows | jq 'map(select(.app=="Terminal" and ."is-native-fullscreen") | .id) | join(" ")' | tr -d '"' | xargs -n1 -I{} yabai -m window {} --toggle native-fullscreen`

**Reply by Demianeen** [reply]
Wow, @koekeishiya! Thanks, never have thought that yabai can do even this)
